RpcSystemAssetTracker/UserSystemAssetCoinOutputsKey.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Neo.IO;
+
+namespace Neo.Plugins
+{
+    public class UserSystemAssetCoinOutputsKey : IComparable<UserSystemAssetCoinOutputsKey>, IEquatable<UserSystemAssetCoinOutputsKey>,
+        ISerializable
+    {
+        public bool IsGoverningToken; // It's either the governing token or the utility token
+        public readonly UInt160 UserAddress;
+        public readonly UInt256 TxHash;
+
+        public int Size => 1 + UserAddress.Size + TxHash.Size;
+
+        public bool Equals(UserSystemAssetCoinOutputsKey other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return IsGoverningToken == other.IsGoverningToken && Equals(UserAddress, other.UserAddress) && Equals(TxHash, other.TxHash);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != this.GetType()) return false;
+            return Equals((UserSystemAssetCoinOutputsKey) obj);
+        }
+
+        public int CompareTo(UserSystemAssetCoinOutputsKey other)
+        {
+            if (ReferenceEquals(this, other)) return 0;
+            if (ReferenceEquals(null, other)) return 1;
+            var isGoverningTokenComparison = IsGoverningToken.CompareTo(other.IsGoverningToken);
+            if (isGoverningTokenComparison != 0) return isGoverningTokenComparison;
+            var userAddressComparison = Comparer<UInt160>.Default.Compare(UserAddress, other.UserAddress);
+            if (userAddressComparison != 0) return userAddressComparison;
+            return Comparer<UInt256>.Default.Compare(TxHash, other.TxHash);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = IsGoverningToken.GetHashCode();
+                hashCode = (hashCode * 397) ^ (UserAddress != null ? UserAddress.GetHashCode() : 0);
```

Comment:
Why 397?

Reply:
It is Prime and my hash code generator plugin seemed to like it.